post_office/utils.py

```diff
@@ -138,14 +138,22 @@ def parse_emails(emails):
     return emails
 
 
-def cleanup_expired_mails(cutoff_date, delete_attachments=True):
+def cleanup_expired_mails(cutoff_date, delete_attachments=True, batch_size=10000):
     """
     Delete all emails before the given cutoff date.
     Optionally also delete pending attachments.
     Return the number of deleted emails and attachments.
     """
     expired_emails = Email.objects.only('id').filter(created__lt=cutoff_date)
-    emails_count, _ = expired_emails.delete()
+    total_deleted_emails = 0
+    while expired_emails:
+        email_ids = expired_emails.values_list('id', flat=True)[:batch_size]
+        
+        # Delete email and incr total_deleted_emails counter
+        _, deleted_data = Email.objects.filter(id__in=email_ids).only('id').delete()
```

Comment:
kenapa pake `only` ya?

ga langsung ambil total_deleted_emails nya dari index yang pertama aja?
```PYTHON
deleted_emails, _ = Email.objects.filter(id__in=email_ids).delete()
total_deleted_emails += deleted_emails
```

Reply:
> why use `only`?, i thinks its not needed for delete operation
> 
> what about this?
> 
> ```python
> deleted_emails, _ = Email.objects.filter(id__in=email_ids).delete()
> total_deleted_emails += deleted_emails
> ```

Actually, I use "only" because referring to the previous code. But yeah, I think it's ok to be removed. For the `deleted_emails` variable, I think it's incorrect because the first element returned from the delete method is the total count for all deleted objects (including Log objects), so it's not an exact deleted email number. That's why I use an explicit deleted email number from the deleted_data['post_office.Email']. Please CMIIW.

Reply:
i see, the logs will be deleted and counted too right

Reply:
Yeah, that's correct.